### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by James-S-Santangelo**
Updating assignment 4. There are a total of 12 marks rather than 8 as in the previous assignments. However, the assignment is still worth 8%.

### CODE REVIEW

**Comment by smile4life** [comment]
I think this looks pretty :fire:

**Comment by smile4life** [comment]
I think the increased difficulty isn't super steep, just enough to encourage improved engagement & push students to actually challenge their understanding? I think your Q3&4 are interesting and important

File: assignment-04.Rmd [hunk 1]
```diff
@@ -1,67 +1,155 @@
...
+    creating this plot for the purpose of evaluating temp temperature, what
+    argument did you adjust and why? (1 mark)
 
-    a. Identify any temperature scores that are two standard deviations away from the mean. Replace them with NA's (missing values). (1 mark)
+    b. What type of variables are temperature and time of day? With this in
+    mind, create a visualization that will help you get a better understanding
+    of the relationship between these variables. (0.5 mark)
 
-    b. Run the following code:
+    c. Create a single box plot that to simultaneously visualise temperature,
```
Comment (aays) [inline assignment-04.Rmd]: s/plot that to/plot to/g

or equivalent

**Comment by aays** [comment]
I think this looks fantastic - I only have a minor copy edit above

File: assignment-04.Rmd [hunk 2]
```diff
@@ -5,83 +5,89 @@ output:
...
     ```
 
-    a. Using what you learned during your last assignment, turn the "activ" column into a categorical variable (factor). Then, create a plot with body temperature on the x-axis. Visualize body temperature separately by whether the beaver is active on the same graph. Hint: we learned about a new useful function that does this during our exploratory data lesson. (1 mark)
-
-    b. Create a dataframe (summary table) that counts the number of temperature recordings that fall into bins with a width of 0.1. Group by day. (1 mark)
+    a. Create a histogram to visualize the distribution of the beavers' body
+    temperatures, separating the temperature data based on the beaver's activity level.
+    (after transforming it into a categorical variable the way you did for your
+    last assignment). Describe the properties of the distribution. When
+    creating this plot for the purpose of evaluating temp temperature, what
```
Comment (ambbb) [inline assignment-04.Rmd]: temp temperature should be body temperature

File: assignment-04.Rmd (hunk 1 above)
Comment (smile4life) [reply assignment-04.Rmd]: Your syntax is :fire: :joy:

File: assignment-04.Rmd [hunk 3]
```diff
@@ -5,83 +5,89 @@ output:
...
     
-        ```{r}
-        set.seed(10)
-        beaver1[sample(1:nrow(beaver1), 20), "temp"] <- NA
-        ```
-
-        Run multiple imputation (set a seed of 100) to replace the missing values. Exclude columns as predictors if necessary. Compare the means of the imputed and original values. (1.5 marks)
+    d. What is one prediction you might make about the relationships among your
+    variables (based on the patterns you observed)? Create a visualization that
+    illustrates your prediction, improving on your other plots in at least one
```
Comment (ambbb) [inline assignment-04.Rmd]: Not sure what "other plot" meant

File: assignment-04.Rmd [hunk 4]
```diff
@@ -5,83 +5,89 @@ output:
...
-    dataset:
+    defenses, herbivores, and pollinators influence the expression of plant
+    floral traits (e.g. flower size). Their experiment had 3 treatments, each
+    with 2 levels: Plant defense (2 levels: defended vs. undefended), herbivory
+    (2 levels: reduced vs. ambient) and pollination (2 levels: open vs.
+    supplemental). These treatments were fully crossed for a total of 8
+    treatment combinations. In each treatment combination, they grew 4
+    individuals from each of 25 plant genotypes for a total of 800 plants (8
+    treatment combinations x 25 genotypes x 4 individuals per genotype). Plants
+    were grown in a common garden at the Koffler Scientific Reserve (UofTs field
```
Comment (ambbb) [inline assignment-04.Rmd]: UofT ' s

File: assignment-04.Rmd [hunk 5]
```diff
@@ -94,94 +100,56 @@ you assignment as a knitted `.pdf` (prefered) or `.html` file.*
...
-              se = sd / sqrt(n)) %>% 
-    ggplot(., aes(x = HCN, y = mean, shape = Herbivory, color = Herbivory)) +
-    geom_errorbar(aes(ymax = mean + se, ymin = mean - se), width = 0.15,
-                  position = position_dodge(width = 0.15)) +
-    geom_point(position = position_dodge(width = 0.15)) +
-    theme_classic()
-```
+    banner petals using a combination of different colours, linetypes, shapes,
+    etc. Feel free to use whatever kind of plot that is appropriate to this kind
+    of data. Avoid overlapping points in the figure. Also include 1 standard
```
Comment (ambbb) [inline assignment-04.Rmd]: include **error bars/bands** with one standard error

File: assignment-04.Rmd [hunk 6]
```diff
@@ -94,94 +100,56 @@ you assignment as a knitted `.pdf` (prefered) or `.html` file.*
...
-Block = 0.003231 / total_var
-
-Genotype
-Genotype_herb
-Block
-```
-    
+    d. After accounting for the fixed effects, what percentage of the variation
+    in banner petal width was explained by each of the random effects in the
+    model? (0.5 marks).
```
Comment (ambbb) [inline assignment-04.Rmd]: Maybe add in "show your work"?

File: assignment-04.Rmd [hunk 7]
```diff
@@ -5,83 +5,89 @@ output:
...
-
-        Hint: you need to make sure there are only two levels to this variable: `"F"` and `"M"`. (0.5 marks)
+    a. Look through the help documentation (?CO2) to understand what each
+    variable means. Which variable(s) do you think would be the $y$ in the GLM
+    model? Which variable(s) would be the $x$? Briefly defend these choices. (1
+    mark)
+
+    b. How much does `uptake` change if `conc` goes up by 10 mL/L? (*Note:* it
+    is intentional that there is no mention of the other variables in the
+    model.) Write out the interpretation as a simple statement of this
```
Comment (ambbb) [inline assignment-04.Rmd]: Not sure what "the model" is referring to?

**Pull request merged**